src/GafferBindings/CompoundNumericPlugBinding.cpp

```diff
@@ -54,6 +56,23 @@ namespace
 {
 
 template<typename T>
+std::string maskedCompoundNumericPlugRepr( const T *plug, unsigned flagsMask )
+{
+	std::string extraArgs = "";
+
+	const IECore::GeometricData::Interpretation interpretation = plug->interpretation();
+	if( interpretation != IECore::GeometricData::None && interpretation != IECore::GeometricData::Color )
```

Comment:
Sorry, I wasn't specific enough in my answer. I was advocating for your first solution, the one that tweaks the OSLShader loading code. As you say, we shouldn't second guess what other clients of OpenImageIOAlgo might want, so should leave OpenImageIOAlgo as it is...